services-camel/src/test/java/edu/unc/lib/cdr/ReplicationProcessorIT.java

```diff
@@ -0,0 +1,193 @@
+import edu.unc.lib.dl.fcrepo4.RepositoryPathConstants;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.util.URIUtil;
+
+/**
+ * @author harring
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/spring-test/test-fedora-container.xml","/spring-test/cdr-client-container.xml"})
+public class ReplicationProcessorIT extends CamelTestSupport {
+    private static final String MIMETYPE = "text/plain";
+
+    @Autowired
+    protected String baseAddress;
+
+    @Autowired
+    protected FcrepoClient client;
+    
+    @Autowired
+    protected Repository repository;
+
+    ReplicationProcessor processor;
+
+    @Mock
+    Exchange exchange;
+    @Mock
+    Message message;
+
+    @Before
+    public void init() throws IOException {
+        PIDs.setRepository(repository);
+        processor = new ReplicationProcessor(repository, "/tmp", 3, 100L);
+        initMocks(this);
+        
+        when(exchange.getIn()).thenReturn(message);
+        when(message.getHeader(CdrBinaryPath)).thenReturn("path/to/bin");
+        when(message.getHeader(CdrBinaryMimeType)).thenReturn(MIMETYPE);
+        when(exchange.getOut()).thenReturn(message);
+    }
+
+    private URI createBaseContainer(String name) throws IOException, FcrepoOperationFailedException {
+        URI baseUri = URI.create(URIUtil.join(baseAddress, name));
+        // Create a parent object to put the binary into
+        try (FcrepoResponse response = client.put(baseUri).perform()) {
+            return response.getLocation();
+        } catch(FcrepoOperationFailedException e) {
+            if (e.getStatusCode() != HttpStatus.SC_CONFLICT) {
+                throw e;
+            }
+            // Ignore duplicate creation of base container
+            return baseUri;
+        }
+    }
+    
+    private URI determineRepositoryPath(URI baseUri) throws IOException, FcrepoOperationFailedException {
+    	    PID parentPid = null;
+        try (FcrepoResponse response = client.post(baseUri).perform()) {
+            parentPid = PIDs.get(response.getLocation());
+        } catch(FcrepoOperationFailedException e) {
+            if (e.getStatusCode() != HttpStatus.SC_CONFLICT) {
+                throw e;
+            }
+        }
+        return parentPid.getRepositoryUri();
+    }
+
+    @Test
+    public void replicateFileFromFedoraTest() throws Exception {
+        // Create a parent object to put the binary into
+        URI contentBase = createBaseContainer(RepositoryPathConstants.CONTENT_BASE);
+        URI binaryUri = determineRepositoryPath(contentBase);
+
+        String bodyString = "Test text";
+        String filename = "test.txt";
+        String checksum = "82022e1782b92dce5461ee636a6c5bea8509ffee";
+        InputStream contentStream = new ByteArrayInputStream(bodyString.getBytes());
+
+        BinaryObject internalObj = repository.createBinary(binaryUri, "binary_test", contentStream, filename, MIMETYPE, checksum, null);
```

Comment:
There's a number of lines that are longer than 120.  I'm not sure what our policy should be on this since we have checkstyle disabled for tests (somewhat annoyingly)